src/Types/Value.purs

```diff
@@ -388,16 +371,51 @@ instance Split Value where
     bimap (flip Value mempty) (flip Value mempty) (split coin)
       <> bimap (Value mempty) (Value mempty) (split nonAdaAsset)
 
--- FIX ME: https://github.com/Plutonomicon/cardano-transaction-lib/issues/193
--- Because our `Value` is different to Plutus, I wonder if this will become an
--- issue.
+newtype PlutusValue = PlutusValue Plutus.Value
+
+derive instance Newtype PlutusValue _
+
+instance FromPlutusType Identity PlutusValue Value where
+  fromPlutusType (PlutusValue plutusValue) =
+    Identity (adaValue <> mkValue mempty nonAdaAssets)
+    where
+    { adaTokenMap, nonAdaTokenMap } =
+      (\x -> { adaTokenMap: x.yes, nonAdaTokenMap: x.no }) <<<
+        partition (\(cs /\ _) -> cs == PlutusValue.adaSymbol) $
+        (unwrap $ PlutusValue.getValue plutusValue)
+
+    adaValue :: Value
+    adaValue = flip mkValue mempty <<< wrap <<< fromMaybe zero $ do
+      adaTokens <- Tuple.snd <$> head adaTokenMap
+      PlutusMap.lookup adaToken adaTokens
+
+    nonAdaAssets :: NonAdaAsset
+    nonAdaAssets = unsafePartial $ fromJust
+      $ mkNonAdaAssetsFromTokenMap
+      $ nonAdaTokenMap <#> \(cs /\ tokens) ->
+          PlutusValue.getCurrencySymbol cs /\ Map.fromFoldable (unwrap tokens)
+
+instance ToPlutusType Identity Value PlutusValue where
```

Comment:
Correct me if I'm wrong but we have to be careful with these conversions. E.g. if we go from PlutusValue -> CSLValue -> PlutusVaue. The final PlutusValue won't necessarily equal the first due to the position Ada appears in the final value compare to the first. In particular, the Ada position can easily move around in the AssocMap? If so, should we add a comment here regarding this, in particular, CSL Value should only be used for CSL related (not To/FromData) and `Plutus.Types.Value` should be used in the `Contract` Monad and we shouldn't convert back and forth inside `Contract`?

For example, I swapped the ordering of the components in `compositeValue` and ran into issues, e.g. the below fails the test. Let me know if I misunderstood something of course!
```
  compositeValues :: Plutus.Value /\ Types.Value
  compositeValues =
    ( fst nonAdaValues <> fst adaValues <>
        PlutusValue.singleton currencySymbol1' tokenName2 amount2
    ) /\
      ( snd nonAdaValues <> snd adaValues <>
          Value.mkValue mempty nonAdaAsset2
      )
```

Reply:
I think this an unfortunate result of `AssocMap` not having an `Ord` on keys which is annoying?